obsreport/obsreport_test.go

```diff
@@ -50,149 +50,174 @@ type testParams struct {
-		default:
-			t.Fatalf("unexpected param: %v", params[i])
+		spans := tt.SpanRecorder.Ended()
+		require.Equal(t, len(params), len(spans))
+
+		var acceptedLogRecords, refusedLogRecords int
+		for i, span := range spans {
+			assert.Equal(t, "receiver/"+receiver.String()+"/LogsReceived", span.Name())
+			switch {
+			case params[i].err == nil:
+				acceptedLogRecords += params[i].items
+				require.Contains(t, span.Attributes(), attribute.KeyValue{Key: obsmetrics.AcceptedLogRecordsKey, Value: attribute.Int64Value(int64(params[i].items))})
+				require.Contains(t, span.Attributes(), attribute.KeyValue{Key: obsmetrics.RefusedLogRecordsKey, Value: attribute.Int64Value(0)})
+				assert.Equal(t, codes.Unset, span.Status().Code)
+			case errors.Is(params[i].err, errFake):
+				refusedLogRecords += params[i].items
+				require.Contains(t, span.Attributes(), attribute.KeyValue{Key: obsmetrics.AcceptedLogRecordsKey, Value: attribute.Int64Value(0)})
+				require.Contains(t, span.Attributes(), attribute.KeyValue{Key: obsmetrics.RefusedLogRecordsKey, Value: attribute.Int64Value(int64(params[i].items))})
+				assert.Equal(t, codes.Error, span.Status().Code)
+				assert.Equal(t, params[i].err.Error(), span.Status().Description)
+			default:
+				t.Fatalf("unexpected param: %v", params[i])
+			}
 		}
-	}
-	require.NoError(t, obsreporttest.CheckReceiverLogs(tt, receiver, transport, int64(acceptedLogRecords), int64(refusedLogRecords)))
+		require.NoError(t, obsreporttest.CheckReceiverLogs(tt, receiver, transport, int64(acceptedLogRecords), int64(refusedLogRecords)))
+	})
 }
 
 func TestReceiveMetricsOp(t *testing.T) {
-	tt, err := obsreporttest.SetupTelemetry()
-	require.NoError(t, err)
-	t.Cleanup(func() { require.NoError(t, tt.Shutdown(context.Background())) })
-
-	parentCtx, parentSpan := tt.TracerProvider.Tracer("test").Start(context.Background(), t.Name())
-	defer parentSpan.End()
+	testTelemetry(t, func(tt obsreporttest.TestTelemetry, useOtel bool) {
+		parentCtx, parentSpan := tt.TracerProvider.Tracer("test").Start(context.Background(), t.Name())
+		defer parentSpan.End()
 
-	params := []testParams{
-		{items: 23, err: errFake},
-		{items: 29, err: nil},
-	}
-	for i, param := range params {
-		rec := NewReceiver(ReceiverSettings{
-			ReceiverID:             receiver,
-			Transport:              transport,
-			ReceiverCreateSettings: tt.ToReceiverCreateSettings(),
-		})
-		ctx := rec.StartMetricsOp(parentCtx)
-		assert.NotNil(t, ctx)
-		rec.EndMetricsOp(ctx, format, params[i].items, param.err)
-	}
-
-	spans := tt.SpanRecorder.Ended()
-	require.Equal(t, len(params), len(spans))
+		params := []testParams{
+			{items: 23, err: errFake},
+			{items: 29, err: nil},
+		}
+		for i, param := range params {
+			rec := NewReceiver(ReceiverSettings{
+				ReceiverID:             receiver,
+				Transport:              transport,
+				ReceiverCreateSettings: tt.ToReceiverCreateSettings(),
+			})
+			if useOtel {
```

Comment:
question: isn't this done already by [the code](https://github.com/open-telemetry/opentelemetry-collector/blob/a9f41a2fbcd713682bc693ce0fba4cd8c948bde7/obsreport/obsreport_receiver.go#L101)? Could we instead adjust the obsreportconfig being used like we do [here](https://github.com/open-telemetry/opentelemetry-collector/blob/bb193a24d0a968027ad48c397899ae3688165756/service/telemetry_test.go#L139)?

Reply:
That is def better! I've created a new registry and passed it during the creation of the Receiver.